ui/article-aem-renderer/src/main/java/org/cru/godtools/article/aem/activity/AemArticleActivity.kt

```diff
@@ -61,12 +74,27 @@ class AemArticleActivity : BaseSingleToolActivity(false) {
      * @return true if the intent was successfully processed, otherwise return false
      */
     private fun processIntent(): Boolean {
-        intent?.extras?.apply {
-            articleUri = getParcelable(EXTRA_ARTICLE) ?: return false
-            return true
+        articleUri = processDeepLink()
+                ?: intent?.extras?.run { getParcelable<Uri>(EXTRA_ARTICLE) }
+                ?: return false
+
+        return true
+    }
+
+    private fun processDeepLink(): Uri? {
+        return when {
+            isValidDeepLink() -> intent?.data?.getQueryParameter(PARAM_URI)?.toUri()?.removeExtension()
+            else -> null
         }
+    }
 
-        return false
+    private fun isValidDeepLink(): Boolean {
+        return intent?.action == Intent.ACTION_VIEW &&
+                intent.data?.run {
+                    (scheme == "http" || scheme == "https") &&
```

Comment:
@frett If I am understanding the .run method, it allows you to use the object that you call it on and and run a returnable method on that data.  For example intent.data is run and scheme, host, and path are all with in intent.data.  Am I correct in my assessment of the its use?